packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,168 @@
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button
+   */
+  @Prop() buttonText: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  }
+
+  disconnectedCallback() {
+    i18next.off('languageChanged');
+  }
+
+  render() {
+    const { 
+      label,
+      name, 
+      buttonText, 
+      required, 
+      multiple,
+      accept,
+      error,
+    } = this;
+    return (
+      <Host>
+        {label && (
+          <label htmlFor="fileInputButton">
+            {label}
+            {required && <span class="required">{i18next.t('required')}</span>}
+          </label>
+        )}
+        <slot></slot>
+        <span id="error-message" role="alert">
+          {error && (
+            <Fragment>
+              <span class="sr-only">{i18next.t('error')}</span> 
+              {error}
+            </Fragment>
+          )}
+        </span>
+        <va-button
+          id="fileInputButton"
+          aria-label={i18next.t(label)}
+          onClick={this.handleButtonClick}
+          secondary
+          text={buttonText ? i18next.t(buttonText) : i18next.t('Upload file')}
```

Comment:
Same here. There's a good change we aren't going to have a translation for whatever the `buttonText` prop happens to be. And I also don't think we have a translation for "Upload file".

![A screenshot of the internationalized story for the file input component. "Required" is in Spanish as "Requerido", but the aria-label and text of the button are in English.](https://user-images.githubusercontent.com/2008881/184673327-9eac39a0-d902-49e7-9422-cd5b528a8f50.png)